File: chapter10/UserApplicationService.cs
public class UserApplicationService{
    // トランザクションによる防御
    public void Register(UserRegisterComamnd command)
    {
        using(var transaciton = connection.BeginTransaction())
        {
            // ユーザー生成処理、重複チェックの後に...
            UserRepository.save(user, transaction); // DBへの保存処理の引数にトランザクションを渡す
            transaction.Commit(); //ここでトランザクションをコミット
        }
    }

    // トランザクションスコープを使った場合
    public void Register(UserRegisterComamnd command)
    {
        using(var transaciton = new TransactionScope())
        {
            // UserRepositoryを使って
            // ユーザー生成処理、重複チェック...
            // DBへの保存処理をしてから...
            transaction.Complete();
        }
    }


}

File: chapter10/UserApplicationService.java
public class UserApplicationService {
    // AOPを使ったパターン
    @Transactional
    public void Register(UserRegisterCommand command) {
        // ユーザー生成処理、重複チェック、
        // UserRepositoryを使ってDBへの保存処理のコード
        // コード本体を書かなくてもメソッド正常終了でコミット、
        // 途中の例外でロールバックを自動でやってくれる
    }
}